tests/test_pois.py

```diff
@@ -0,0 +1,274 @@
+
+        r1 = requests.get('http://127.0.0.1:5000/pois?map_year={}'.format(map_year))
+        response1 = r1.json()
+
+        self.assertEqual(response1['code'], 200)
+
+        for i in response1['result']['pois']:
+            self.test0_2_get_poi_by_id(poi_id = i['_id'])
+            self.assertEqual(i['map_year'], map_year)
+
+    def test0_3_get_pois_by_story_id(self):
+        story_ids = poi_create_complete['story_ids']
+
+        #get request by story_id
+
+        for curr_story_id in story_ids:
+            r2 = requests.get('http://127.0.0.1:5000/pois?story_id={}'.format(curr_story_id))
+            response2 = r2.json()
+            self.assertEqual(response2['code'], 200)
+            #created_object_included is used to determine if the POI we previously created is included
+            #in the list of POIs returned after we get request based on story_ids
+            created_object_included = False
+            for curr_poi in response2['result']['pois']:
+                if curr_poi['name'] == poi_create_complete['name']:
+                    created_object_included = True
+            self.assertTrue(created_object_included)
+
+    def test0_4_update_poi(self):
+        poi_id = int(os.environ.get('POI_ID'))
+        r = requests.put('http://127.0.0.1:5000/pois/{}'.format(poi_id), json=poi_update_complete)
+        response = r.json()
+        self.assertEqual(response['code'], 200)
+        self.assertEqual(response['result']['poi']['name'], poi_update_complete['name'])
+        # self.assertEqual(response['result']['poi']['date'], poi_update_complete['date'])
+        self.assertEqual(response['result']['poi']['description'], poi_update_complete['description'])
+        for i in range(len(response['result']['poi']['links'])):
+            self.assertEqual(response['result']['poi']['links'][i]['link_url'], poi_update_complete['links'][i]['link_url'])
+            self.assertEqual(response['result']['poi']['links'][i]['display_name'], poi_update_complete['links'][i]['display_name'])
+        for i in range(len(response['result']['poi']['media'])):
+            self.assertEqual(response['result']['poi']['media'][i]['content_url'], poi_update_complete['media'][i]['content_url'])
+            self.assertEqual(response['result']['poi']['media'][i]['caption'], poi_update_complete['media'][i]['caption'])
+        response_story_ids = [i['_id'] for i in response['result']['poi']['stories']]
+        self.assertEqual(sorted(response_story_ids), sorted(poi_update_complete['story_ids']))
+
+    def test0_5_delete_poi(self):
+        poi_id = int(os.environ.get('POI_ID'))
+        r = requests.delete('http://127.0.0.1:5000/pois/{}'.format(poi_id))
+        response = r.json()
+        self.assertEqual(response['code'], 200)
+
+        r2 = requests.get('http://127.0.0.1:5000/pois/{}'.format(poi_id))
+        response2 = r2.json()
+        self.assertEqual(response2['code'], 404)
+
+    def test1_1_permissible(self):
```

Comment:
similar comments to what i said in `test0_1_create_poi`

Reply:
also keep in mind that you can remove some of these checks by removing the optional fields

Reply:
also^2, do you think the `permissible` tests are even necessary? they are pretty much covered by the `complete` tests